--- a/src/or/microdesc.c
+++ b/src/or/microdesc.c
@@ -344,7 +344,7 @@ microdesc_cache_reload(microdesc_cache_t
 
 /** By default, we remove any microdescriptors that have gone at least this
  * long without appearing in a current consensus. */
-#define TOLERATE_MICRODESC_AGE (7*24*60*60)
+#define TOLERATE_MICRODESC_AGE (24*60*60)
 
 /** Remove all microdescriptors from <b>cache</b> that haven't been listed for
  * a long time.  Does not rebuild the cache on disk.  If <b>cutoff</b> is
